packages/fx-core/src/question/other.ts

```diff
@@ -880,12 +881,136 @@ export function addAuthActionQuestion(): IQTreeNode {
         data: authNameQuestion(),
       },
       {
-        data: apiAuthQuestion(true),
+        data: addAuthActionAuthTypeQuestion(),
       },
+      oauthParametersQuestion(),
+      apiKeyParameterQuestion(),
     ],
   };
 }
 
+export function addAuthActionAuthTypeQuestion(): SingleSelectQuestion {
+  return {
+    type: "singleSelect",
+    name: QuestionNames.ApiAuth,
+    title: getLocalizedString("core.createProjectQuestion.apiMessageExtensionAuth.title"),
+    placeholder: getLocalizedString(
+      "core.createProjectQuestion.apiMessageExtensionAuth.placeholder"
+    ),
+    cliDescription: "The authentication type for the API.",
+    staticOptions: AddAuthActionAuthTypeOptions.all(),
+    default: AddAuthActionAuthTypeOptions.bearerToken().id,
+  };
+}
+
+export function oauthParametersQuestion(): IQTreeNode {
+  return {
+    data: oauthAuthorizationUrlQuestion(),
+    condition: (inputs: Inputs) => {
+      return inputs[QuestionNames.ApiAuth] === AddAuthActionAuthTypeOptions.oauth().id;
+    },
+    children: [
+      {
+        data: oauthTokenUrlQuestion(),
```

Comment:
If it is microsoft entra sso, then token url and refresh url, auth url is not needed